icu4c/source/common/cjmlbe.cpp

```diff
@@ -0,0 +1,462 @@
+    }
+    if (w3 != INVALID) {
+        arr[0] = w3;
+        arr[1] = -1;
+        featureList->adoptElement(concatChar("UW3:", arr, status), status);
+    }
+    if (w4 != INVALID) {
+        arr[0] = w4;
+        arr[1] = -1;
+        featureList->adoptElement(concatChar("UW4:", arr, status), status);
+    }
+    if (w5 != INVALID) {
+        arr[0] = w5;
+        arr[1] = -1;
+        featureList->adoptElement(concatChar("UW5:", arr, status), status);
+    }
+    if (w6 != INVALID) {
+        arr[0] = w6;
+        arr[1] = -1;
+        featureList->adoptElement(concatChar("UW6:", arr, status), status);
+    }
+    if (w2 != INVALID && w3 != INVALID) {
+        arr[0] = w2;
+        arr[1] = w3;
+        arr[2] = -1;
+        featureList->adoptElement(concatChar("BW1:", arr, status), status);
+    }
+    if (w3 != INVALID && w4 != INVALID) {
+        arr[0] = w3;
+        arr[1] = w4;
+        arr[2] = -1;
+        featureList->adoptElement(concatChar("BW2:", arr, status), status);
+    }
+    if (w4 != INVALID && w5 != INVALID) {
+        arr[0] = w4;
+        arr[1] = w5;
+        arr[2] = -1;
+        featureList->adoptElement(concatChar("BW3:", arr, status), status);
+    }
+    if (w1 != INVALID && w2 != INVALID && w3 != INVALID) {
+        arr[0] = w1;
+        arr[1] = w2;
+        arr[2] = w3;
+        arr[3] = -1;
+        featureList->adoptElement(concatChar("TW1:", arr, status), status);
+    }
+    if (w2 != INVALID && w3 != INVALID && w4 != INVALID) {
+        arr[0] = w2;
+        arr[1] = w3;
+        arr[2] = w4;
+        arr[3] = -1;
+        featureList->adoptElement(concatChar("TW2:", arr, status), status);
+    }
+    if (w3 != INVALID && w4 != INVALID && w5 != INVALID) {
+        arr[0] = w3;
+        arr[1] = w4;
+        arr[2] = w5;
+        arr[3] = -1;
+        featureList->adoptElement(concatChar("TW3:", arr, status), status);
+    }
+    if (w4 != INVALID && w5 != INVALID && w6 != INVALID) {
+        arr[0] = w4;
+        arr[1] = w5;
+        arr[2] = w6;
+        arr[3] = -1;
+        featureList->adoptElement(concatChar("TW4:", arr, status), status);
+    }
+    if (b1 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("UB1:", b1, status), status);
+    }
+    if (b2 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("UB2:", b2, status), status);
+    }
+    if (b3 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("UB3:", b3, status), status);
+    }
+    if (b4 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("UB4:", b4, status), status);
+    }
+    if (b5 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("UB5:", b5, status), status);
+    }
+    if (b6 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("UB6:", b6, status), status);
+    }
+    if (b2 != UnicodeString(INVALID) && b3 != UnicodeString(INVALID)) {
```

Comment:
UnicodeString(INVALID) is used many times, why don't you just create one 
UnicodeString invalid(INVALID);
and use invalid below in the comparision line

Reply:
Done. Thanks for the suggestion.